inference-engine/ie_bridges/c/src/CMakeLists.txt

```diff
@@ -21,6 +21,12 @@ target_include_directories(${TARGET_NAME} PUBLIC "${InferenceEngine_C_API_SOURCE
 
 add_cpplint_target(${TARGET_NAME}_cpplint FOR_TARGETS ${TARGET_NAME})
 
+# Workaround to avoid warnings caused with bug in the avx512intrin.h of GCC5
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND
+   (CMAKE_CXX_COMPILER_VERSION VERSION_LESS_EQUAL 5.5))
+    set_target_properties(${TARGET_NAME} PROPERTIES LINK_FLAGS_RELEASE "-Wno-error=maybe-uninitialized -Wno-error=uninitialized -Wno-uninitialized -Wno-maybe-uninitialized")
+endif()
```

Comment:
why do we need this for C bridges?

Reply:
It turns out that C-bridges links fluid or preprocessing (don't remember exactly). But I remember that during linking gcc-5 generate warnings related with bug (poorly written intrinsic) in avx512fintrin.h header of gcc-5.

Reply:
no, C API does not link preprocessing.

Reply:
[2020-05-19T22:58:32.313Z] [ 70%] Built target inference_engine_c_api
[2020-05-19T22:58:32.313Z] /home/jenkins/agent/workspace/private-ci/ie/build-linux-ubuntu16/b/repos/closed-dldt/inference-engine/thirdparty/ocv/opencv_hal_avx512.hpp: In function 'v_gather_chan':
[2020-05-19T22:58:32.313Z] /usr/lib/gcc/x86_64-linux-gnu/5/include/avx512fintrin.h:8512:50: error: '__Y' is used uninitialized in this function [-Werror=uninitialized]
[2020-05-19T22:58:32.313Z]    return (__m512i) __builtin_ia32_alignd512_mask ((__v16si) __A,
[2020-05-19T22:58:32.313Z]                                                   ^
[2020-05-19T22:58:32.313Z] /usr/lib/gcc/x86_64-linux-gnu/5/include/avx512fintrin.h:135:11: note: '__Y' was declared here
[2020-05-19T22:58:32.313Z]    __m512i __Y = __Y;
[2020-05-19T22:58:32.313Z]            ^

Reply:
I think we need to concrete that this is bug not of GCC5 but of GCC on Ubuntu?

Reply:
May be. But this is not so important in this case. What is more important is what to do with it.

Reply:
You here check for compiler version - does it work with others?

Reply:
Discussed offline - no questions more.